src/Symfony/Component/Security/Http/Firewall/ChannelListener.php

```diff
@@ -47,6 +47,9 @@ public function handle(GetResponseEvent $event)
         if ('https' === $channel && !$request->isSecure()) {
             if (null !== $this->logger) {
                 $this->logger->info('Redirecting to HTTPS.');
+                if ('https' === $request->headers->get('X-Forwarded-Proto')) {
+                    $this->logger->debug('Possible redirect loop - did you set "trusted_proxies" correctly?');
```

Comment:
What about `warn`?

Reply:
This log message could be triggered by anybody who sends HTTP requests to the server with that header:

```bash
curl http://www.example.com -H 'X-Forwarded-Proto: https'
```

If the site is configured to log messages with a minimum severity of `warn` or higher then their logs could potentially be flooded with "false positives".  Using `debug()` would prevent that from happening unless the site is configured to log `debug` messages (which I'd assume somebody might do if they had an unexpected redirect loop and were looking for clues).

That being said, I'm not strongly opposed to using a different log level if others feel that it is more appropriate.

Reply:
since there is already an info level message, we could merge the new with the existing one:
`$this->logger->info('Redirecting to HTTPS ("X-Forwarded-Proto" is set to "https" - did you set "trusted_proxies" correctly?).');`

Reply:
Good idea!  I've made it so that only one message will ever be logged.